app/models/ability.rb

```diff
@@ -35,7 +35,42 @@ def initialize(user)
     #   end
 
     user ||= User.new # guest user (not logged in)
+    id = user.id
+
     # All users can only update their own user attributes
-    can :update, User, id: user.id
+    alias_action :create, :read, :update, :destroy, :to => :crud
+
+    can :crud, Team
+    can :update, User, id: id
+
+    can :assign_ownership, Team, Team do |team|
+      team.owners.include? user
+    end
+
+    can :delete_ownership, Team, Team do |team|
+      team.owners.include? user
+      team.owners.length > 1
+    end
+
+    can :delete_membership, Team, Team do |team|
+      team.owners.include? user and Ability.num_owners(team, team_member) > 0
+    end
+
+    can :delete_membership, Team, Team do |team|
+      Ability.num_owners(team, team_member) > 0 and Integer(id) == Integer(team_member)
+    end
   end
+
+  private
+
+  def self.num_owners(team, team_member)
```

Comment:
Can you maybe find a more descriptive name for this method? Maybe something like `number_of_owners_after_deleting_member`?

Reply:
oh! i thought i had done this as originally proposed by @lheilmann. will rename.